File: src/platform/android/src/com/technegames/tappyplane/Logger.java
package com.technegames.tappyplane;

import android.util.Log;

/**
 * @author stephen gowen <stephen.gowen@chexar.com>
 */
public final class Logger
{
    /**
     * @param loggingLevel
     *            "ERROR" for error "WARN" for error and warning "INFO" for
     *            error, warning, and info "DEBUG" for error, warning, info, and
     *            debug Any other String will result in NO LOGGING
     */
    public static void initializeLoggingLevel(String loggingLevel)
    {
        if (loggingLevel.equalsIgnoreCase("DEBUG"))
        {
            Logger.loggingLevel = 4;
        }
        else if (loggingLevel.equalsIgnoreCase("INFO"))
        {
            Logger.loggingLevel = 3;
        }
        else if (loggingLevel.equalsIgnoreCase("WARN"))
        {
            Logger.loggingLevel = 2;
        }
        else if (loggingLevel.equalsIgnoreCase("ERROR"))
        {
            Logger.loggingLevel = 1;
        }
        else
        {
            Logger.loggingLevel = 0;
        }
    }

    public static boolean isErrorEnabled()
    {
        return loggingLevel >= 1;
    }

    public static boolean isWarnEnabled()
    {
        return loggingLevel >= 2;
    }

    public static boolean isInfoEnabled()
    {
        return loggingLevel >= 3;
    }

    public static boolean isDebugEnabled()
    {
        return loggingLevel >= 4;
    }

    private static int loggingLevel = 4;

    private final String TAG;

    public Logger(Class<?> clazz)
    {
        this.TAG = clazz.getSimpleName();
    }

    public void debug(String message)
    {
        if (loggingLevel >= 4)
        {
            Log.d(TAG, "[DEBUG] " + message);
        }
    }

    public void info(String message)
    {
        if (loggingLevel >= 3)
        {
            Log.i(TAG, "[INFO] " + message);
        }
    }

    public void warn(String message)
    {
        if (loggingLevel >= 2)
        {
            Log.w(TAG, "[WARN] " + message);
        }
    }

    public void warn(String message, Exception exception)
    {
        if (loggingLevel >= 2)
        {
            Log.w(TAG, "[WARN] " + message, exception);
        }
    }

    public void error(String message)
    {
        if (loggingLevel >= 1)
        {
            Log.e(TAG, "[ERROR] " + message);
        }
    }

    public void error(String message, Exception exception)
    {
        if (loggingLevel >= 1)
        {
            Log.e(TAG, "[ERROR] " + message, exception);
        }
    }

    public void error(String message, Throwable cause)
    {
        if (loggingLevel >= 1)
        {
            Log.e(TAG, "[ERROR] " + message, cause);
        }
    }
}

File: src/platform/ios/Logger.m
//
//  Logger.m
//  tappyplane
//
//  Created by Stephen Gowen on 2/27/14.
//  Copyright (c) 2014 Gowen Game Dev. All rights reserved.
//

#import "Logger.h"

#define LOG_DEBUG @"DEBUG"
#define LOG_INFO @"INFO"
#define LOG_WARN @"WARN"
#define LOG_ERROR @"ERROR"

static int loggingLevel = 4;

@implementation Logger

+ (void)initializeLoggingLevelFromNSString:(NSString *)loggingLevelName
{
    if([loggingLevelName isEqualToString:LOG_DEBUG])
    {
        loggingLevel = 4;
    }
    else if([loggingLevelName isEqualToString:LOG_INFO])
    {
        loggingLevel = 3;
    }
    else if([loggingLevelName isEqualToString:LOG_WARN])
    {
        loggingLevel = 2;
    }
    else if([loggingLevelName isEqualToString:LOG_ERROR])
    {
        loggingLevel = 1;
    }
    else
    {
        loggingLevel = 0;
    }
}

- (id)initWithClass:(Class)c
{
    self = [super init];
    
    if(self)
    {
        TAG = NSStringFromClass(c);
    }
    
    return self;
}

- (void)debug:(NSString *)message
{
    if (loggingLevel >= 4)
    {
        NSLog(@"%@ [%@] %@", TAG, LOG_DEBUG, message);
    }
}

- (void)info:(NSString *)message
{
    if (loggingLevel >= 3)
    {
        NSLog(@"%@ [%@] %@", TAG, LOG_INFO, message);
    }
}

- (void)warn:(NSString *)message
{
    if (loggingLevel >= 2)
    {
        NSLog(@"%@ [%@] %@", TAG, LOG_WARN, message);
    }
}

- (void)error:(NSString *)message
{
    if (loggingLevel >= 1)
    {
        NSLog(@"%@ [%@] %@", TAG, LOG_ERROR, message);
    }
}

- (void)error:(NSString *)message withException:(NSException *)e
{
    if (loggingLevel >= 1)
    {
        NSLog(@"%@ [%@] %@, Exception: %@, Stack Trace: %@", TAG, LOG_ERROR, message, [e description], [e callStackSymbols]);
    }
}

- (void)error:(NSString *)message withError:(NSError *)e
{
    if (loggingLevel >= 1)
    {
        NSLog(@"%@ [%@] %@, Error: %@", TAG, LOG_ERROR, message, [e description]);
    }
}

+ (bool)isErrorEnabled
{
    return loggingLevel >= 1;
}

+ (bool)isWarnEnabled
{
    return loggingLevel >= 2;
}

+ (bool)isInfoEnabled
{
    return loggingLevel >= 3;
}

+ (bool)isDebugEnabled
{
    return loggingLevel >= 4;
}

@end
